[FIX] purchase_landed_cost: error on deleting cost lines

cubells opened the pull request:
Before this pr:
If user try to delete some cost line and this cost line is referenced in any expense line into the field 'affected lines', a not _sensitive_ error is shown to user.

After this pr:
Error is catched and an _clear_ message is shown to user.

cc @Tecnativa

cubells committed:
[FIX] purchase_landed_cost: error on deleting cost lines

pedrobaeza commented:
Can't this be handled through ondelete="cascade"?

cubells commented:
@pedrobaeza 
cost_lines has `ondelete` parameter: https://github.com/Tecnativa/purchase-workflow/blob/11.0-product_landed_cost_fix_error/purchase_landed_cost/models/purchase_cost_distribution.py#L120

and error is raised when deleting a cost line without remove reference in expense lines.

pedrobaeza commented:
That ondelete doesn't serve. It should be on the `many2one` part, not on the `one2many`

cubells commented:
I've tested `ondelete` on`affected_lines` field and it does not work, but `affected_lines` field is Many2many.

cubells committed:
[IMP] purchase_landed_cost: Initial tests

pedrobaeza committed:
[FIX+IMP] purchase_landed_cost: Fix price update + improve tests

* Previously, price update takes the price from involved quants, but they can be split
  or changed in other process, corrupting correct new price. Now the price is computed
  from the difference over prices, adding or removing price surcharge for getting
  the final computation.
* Tests adapted and new ones added for checking extra use cases.

pedrobaeza commented:
@Roodin were you finally using this?

rafaelbn commented on purchase_landed_cost/__manifest__.py:
```diff
@@ -5,12 +5,12 @@
+              u"Joaquín Gutierrez,"
+              u"Odoo Community Association (OCA)",
     'category': 'Purchase Management',
     'website': 'http://www.odoomrp.com',
```
Change to https://github.com/OCA/purchase-workflow
```suggestion
    'website': 'https://github.com/OCA/purchase-workflow',
```
As said in https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L11

rafaelbn commented:
Are we going to improve  codecov/patch — 76.47% of diff hit (target 81.1%) ?

Can we merge ?

pedrobaeza commented:
Yes, merging, the coverage is good on the project and module, but as this is reducing code, that's why it's red.
